packages/button/test/button.test.js

```diff
@@ -0,0 +1,161 @@
+import { expect } from '@esm-bundle/chai';
+import sinon from 'sinon';
+import {
+  arrowDownKeyDown,
+  enterKeyDown,
+  enterKeyUp,
+  fixtureSync,
+  isIOS,
+  mousedown,
+  mouseup,
+  spaceKeyDown,
+  spaceKeyUp,
+  touchend,
+  touchstart
+} from '@vaadin/testing-helpers';
+import { FlattenedNodesObserver } from '@polymer/polymer/lib/utils/flattened-nodes-observer.js';
+import { Button } from '../vaadin-button.js';
+
+customElements.define(Button.is, Button);
+
+describe('vaadin-button', () => {
+  let element;
+
+  describe('custom element definition', () => {
+    let tagName;
+
+    beforeEach(() => {
+      element = fixtureSync('<vaadin-button></vaadin-button>');
+      tagName = element.tagName.toLowerCase();
+    });
+
+    it('should be defined in custom element registry', () => {
+      expect(customElements.get(tagName)).to.be.ok;
+    });
+
+    it('should have a valid static "is" getter', () => {
+      expect(customElements.get(tagName).is).to.equal(tagName);
+    });
+
+    it('should have a valid version number', () => {
+      expect(customElements.get(tagName).version).to.match(/^(\d+\.)?(\d+\.)?(\d+)(-(alpha|beta|rc)\d+)?$/);
+    });
+  });
+
+  describe('default', () => {
+    let button, label;
+
+    beforeEach(() => {
+      element = fixtureSync('<vaadin-button>Vaadin <i>Button</i></vaadin-button>');
+      button = element.shadowRoot.querySelector('button');
+      label = element.shadowRoot.querySelector('[part=label]');
+    });
+
+    it('should have [type=button] on the native button', () => {
+      expect(button.getAttribute('type')).to.be.eql('button');
+    });
+
+    it('should define the button label using light DOM', () => {
+      const children = FlattenedNodesObserver.getFlattenedNodes(label);
+      expect(children[1].textContent).to.be.equal('Vaadin ');
+      expect(children[2].outerHTML).to.be.equal('<i>Button</i>');
+    });
+
+    it('should fire the click event', () => {
+      const spy = sinon.spy();
+      element.addEventListener('click', spy);
+      element.click();
+      expect(spy.calledOnce).to.be.true;
+    });
```

Comment:
To me this test does not make much sense. Looks like it has been there since the [initial implementation](https://github.com/vaadin/vaadin-button/commit/be9be2b747df90ec3b38bd94efc58045248bbf87#diff-7ef320f40e98e8da602ffe199ca9218264362aded5ac29951d251e21175c7c60R42). Let's remove it as there is no specific logic to e.g. dispatch click events manually in the component.

Reply:
Done.